### DISCUSSION TITLE
What's the most helpful way a person could start contributing?

### DISCUSSION THREAD

**Discussion by digamma-fen** [helix-editor/helix#10719] [General]
I use helix for all my coding, and I love it, and I want to help.

There's several issues I've almost started to work on, just to realize that there's a PR up waiting for review.
It looks to me like the bottleneck is on reviews not on code contribution (which a great place to be in I think?).
But it leaves me in an odd place where im not sure if starting to contribute now would be counter-productive to getting work through the review pipeline.
Any thoughts?

**Comment by the-mikedavis** [comment]
This is a good question, that's for opening it up.

I see submitting PRs as a separate concept than giving back / helping out - submitting PRs is about sharing a change you'd like to see with others. Often that's aligned with helping out but not necessarily always. So I would discourage you from sending PRs only because you want to give back / help out. Instead I would recommend changing how things work locally in a fork and submitting that as a PR if you think it's a good fit (fixes something or is not too burdensome to maintain and not language/workflow-specific, etc.).

In terms of donating your time to help out the project: one thing I'd love to see is people helping out with triaging things that might be bugs. https://github.com/helix-editor/helix/issues/10689 for example is definitely a bug but needs some further investigation to find the right place to put the fix. There's also a lot of [troubleshooting discussions](https://github.com/helix-editor/helix/discussions/categories/troubleshooting) that are less obviously bugs in Helix and need more investigation. Sometimes the problem ends up being in the language server or tree-sitter grammar, or just with the user's config or local setup. Debugging these things can take a bunch of time and I usually give review a higher priority myself, but it's a good way to get familiar with the plumbing of language servers and Helix if you can spare the time/effort to dive in. Hanging out in the Matrix channel and answering questions is also a good way to get involved and help out.

**Comment by markstos** [comment]
I'm outsider here, but I have experience contributing and maintaining a number of open source projects. I agree with @the-mikedavis that a lot of what open source projects lack is not more PRs-- that's the most common way that people are interested to contribute. Providing support, triaging, improving docs and other tasks are very welcome because they free up time for the maintainers to actually work on coding and dealing with the PR backlog. 

A concrete thing you could do is subscribe to "new issue notifications". Check out new issues and discussions as they come in.  Have people provided enough information for a good bug report? Can you reproduce it or figure out the answer?
